large_image/tilesource/jupyter.py

```diff
@@ -393,19 +399,55 @@ def from_map(self, coordinate: Union[List[float], Tuple[float, float]]) -> Tuple
             return transf.transform(x, y)
         return x, self._metadata['sizeY'] - y
 
-    def update_frame(self, event, **kwargs):
-        frame = int(event.get('new'))
+    def update_frame(self, frame, style, **kwargs):
         if self._layer:
-            if 'frame=' in self._layer.url:
-                self._layer.url = re.sub(r'frame=(\d+)', f'frame={frame}', self._layer.url)
-            else:
-                if '?' in self._layer.url:
-                    self._layer.url = self._layer.url.replace('?', f'?frame={frame}&')
-                else:
-                    self._layer.url += f'?frame={frame}'
-
+            parsed_url = urlparse(self._layer.url)
+            scheme = parsed_url.scheme
+            netloc = parsed_url.netloc
+            path = parsed_url.path
+            query = parsed_url.query
+            query = {k: v[0] for k, v in parse_qs(query).items()}
+            query.update(dict(
+                frame=frame,
+                style=json.dumps(style),
+            ))
+            query_string = urlencode(query, quote_via=quote, safe='{}')
+            self._layer.url = f'{scheme}://{netloc}{path}?{query_string}'
```

Comment:
scheme could be `''`, at which point this will fail.  It is better to use urlunparse to make sure it is done right.

Reply:
Ok, changed in 724c7f9